src/Type/Union/MenuItemObjectUnion.php

```diff
@@ -0,0 +1,46 @@
+<?php
+namespace WPGraphQL\Type;
+
+use WPGraphQL\TypeRegistry;
+use WPGraphQL\Types;
+
+$args = [
+	'show_in_graphql' => true,
+];
+
+$possible_types = [];
+
+// Add post types that are allowed in WPGraphQL.
+foreach ( get_post_types( $args ) as $type ) {
+	$possible_types[ $type ] = Types::post_object( $type );
+}
+
+// Add taxonomies that are allowed in WPGraphQL.
+foreach ( get_taxonomies( $args ) as $type ) {
+	$possible_types[ $type ] = Types::term_object( $type );
+}
+
+// Add the custom link type (which is just a menu item).
+$possible_types['MenuItem'] = Types::menu_item();
+
+register_graphql_union_type( 'MenuItemObjectUnion', [
```

Comment:
Is it possible to alias union types? One of my regrets in my Menus PR was naming this `MenuItemObjectUnion`. That's not actually what it is and it's more broadly useful than that since it's effectively a union of all supported post types, users, and terms.

If we can't alias it later it might be nice to get it changed before `0.1.0`.

Reply:
@chriszarate In my head it makes sense to still have a `MenuItemObjectUnion` as we _should_ (if we aren't already) probably only respect types that are registered with `show_in_nav_menus => true`, eh?

I do think there is room for a Union of _all_ possible PostTypeObjects and TermObjects, but might make sense to be _another_ union, not this one?

Reply:
I do see the code comment here about this topic: https://github.com/wp-graphql/wp-graphql/blob/61786c27f1101aff61fd80167707dd056de94189/src/Type/Union/MenuItemObjectUnionType.php#L66-L68 🤔

Reply:
So, since the Schema itself is queryable, I do think there is value in restricting this to types that are registered with `show_in_nav_menus => true`, because GraphQL can serve as a debug tool just as much as it can serve as an API for clients. 

For example, as an admin on a WordPress site (or even a UI developer), I may want to quickly find out what Types are registered to show in nav menus, and a quick query to do that would be: 

```
{
  __type(name: "MenuItemObjectUnion") {
    possibleTypes {
      name
      description
      kind
    }
  }
}
```

Reply:
Here's an example of that Query run with and without `'show_in_nav_menus' => true`

You can see that "attachments / mediaItems)" are not registered to `show_in_nav_menus`

I think I'd argue that this Type _should_ stay, and we should add `show_in_nav_menus => true` to it, and introduce another Type which supports an intersection of all PostTypeObjects + TermObjects

![screen shot 2018-10-22 at 10 34 20 am](https://user-images.githubusercontent.com/1260765/47305275-2f721480-d5e6-11e8-8d0e-dbde4fd1ee08.png)
![screen shot 2018-10-22 at 10 34 33 am](https://user-images.githubusercontent.com/1260765/47305276-2f721480-d5e6-11e8-85e7-3357ebf2cf30.png)

Reply:
Makes sense.